File: chapter 08/recipe 8.5/DniScJITVB/DniScNotJITObj.vb
Imports System.Runtime.InteropServices
Imports System.EnterpriseServices

<ClassInterface(ClassInterfaceType.None)> _
Public Class DniScNotJITObj
    Inherits ServicedComponent
    Implements IJITMethods

    Private m_Number As Integer

    Public Sub AddNumber(ByVal number As Integer) _
        Implements IJITMethods.AddNumber

        m_Number += number

    End Sub

    Public Function GetTotal() As Integer _
        Implements IJITMethods.GetTotal

        Return m_Number

    End Function

    Public Function AddSomeNumbers(ByVal numA As Integer, _
        ByVal numB As Integer) As Integer _
        Implements IJITMethods.AddSomeNumbers

        Return numA + numB

    End Function

End Class

File: chapter 08/recipe 8.5/DniScJIT/DniScNotJITObj.cs
using System;
using System.Runtime.InteropServices;
using System.EnterpriseServices;

namespace DniScJIT
{
	[ClassInterface(ClassInterfaceType.None)]
	public class DniScNotJITObj 
		: ServicedComponent, IJITMethods
	{
		private int m_Number;

		public void AddNumber(int number)
		{
			m_Number += number;
		}

		public int GetTotal()
		{
			return m_Number;
		}

		public int AddSomeNumbers(int numA, int numB)
		{
			return numA + numB;
		}
	}
}
